etl/steps/data/garden/ember/2022-12-13/yearly_electricity.py

```diff
@@ -0,0 +1,387 @@
+    units = table.columns.get_level_values(1).tolist()
+    variable_units = [f"{variable} ({unit})" for variable, unit in table.columns]
+
+    assert len(variable_units) == len(units) == len(variables)
+
+    # Collapse the two column levels into one, with the naming "variable (unit)" (except for country and year, that
+    # have no units).
+    table.columns = variable_units
+
+    # Add region aggregates.
+    aggregates = {column: "sum" for column in SUM_AGGREGATES if column in table.columns}
+
+    table = add_region_aggregates(
+        data=table.reset_index(),
+        index_columns=["country", "year"],
+        regions=REGIONS,
+        aggregates=aggregates,
+    )
+
+    return table
+
+
+def process_electricity_generation(df: pd.DataFrame) -> catalog.Table:
+    """Create table with processed data of category "Electricity generation".
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Data in long format for all categories, after harmonizing country names.
+
+    Returns
+    -------
+    table : catalog.Table
+        Table of processed data for the given category.
+
+    """
+    # Prepare wide table.
+    table = prepare_wide_table(df=df, category="Electricity generation")
+
+    # Recalculate the share of electricity generates for region aggregates.
+    for column in table.columns:
+        if "(%)" in column:
+            # Find corresponding column with units instead of percentages.
+            value_column = column.replace("(%)", "(TWh)")
+            if value_column not in table.columns:
+                raise ValueError(f"Column {value_column} not found.")
+            # Select only regions.
+            select_regions = table["country"].isin(list(REGIONS))
+            table.loc[select_regions, column] = table[value_column] / table["Total Generation (TWh)"] * 100
+
+    return table
+
+
+def process_electricity_demand(df: pd.DataFrame) -> catalog.Table:
+    """Create table with processed data of category "Electricity demand".
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Data in long format for all categories, after harmonizing country names.
+
+    Returns
+    -------
+    table : catalog.Table
+        Table of processed data for the given category.
+
+    """
+    # Prepare wide table.
+    table = prepare_wide_table(df=df, category="Electricity demand")
+
+    # Add population to data
+    table = add_population(df=table, warn_on_missing_countries=False)
+
+    # Recalculate demand per capita.
+    # We could do this only for region aggregates (since they do not have per capita values),
+    # but we do this for all countries, to ensure per-capita variables are consistent with our population data.
+    table["Demand per capita (kWh)"] = (
+        pd.DataFrame(table)["Demand (TWh)"] * TWH_TO_KWH / pd.DataFrame(table)["population"]
+    )
+
+    # Delete the original demand per capita column.
+    table = table.drop(columns="Demand per capita (MWh)")
```

Comment:
For consistency with other uses that you make of `pd.DataFrame.drop`, I'd provide the value for argument `columns` as a list, i.e.

```python
table = table.drop(columns=["Demand per capita (MWh)"])
```